geometry/bounding_box.cpp

```diff
@@ -14,14 +15,24 @@ BoundingBox::BoundingBox(vector<PointD> const & points)
 
 void BoundingBox::Add(double x, double y)
 {
-  m_minX = min(m_minX, x);
-  m_minY = min(m_minY, y);
-  m_maxX = max(m_maxX, x);
-  m_maxY = max(m_maxY, y);
+  m_min.x = min(m_min.x, x);
+  m_min.y = min(m_min.y, y);
+  m_max.x = max(m_max.x, x);
+  m_max.y = max(m_max.y, y);
 }
 
 bool BoundingBox::HasPoint(double x, double y) const
 {
-  return x >= m_minX && x <= m_maxX && y >= m_minY && y <= m_maxY;
+  return x >= m_min.x && x <= m_max.x && y >= m_min.y && y <= m_max.y;
+}
+
+string DebugPrint(BoundingBox const & bbox)
```

Comment:
Не хочешь сделать через визир, коль скоро он уже есть в BoundingBox?

Reply:
Комментарий относится и DebugPrint классов, в которых есть визиторы.

Reply:
Well, only if I'll find a spare time.

Reply:
Ok, but while implementing visitors for transit I implemented a visitor for DebugPrint(s). I like the idea of using visitors instead of serialize/deserialize/debugprint/and so on.